src/Spago/FetchPackage.hs

```diff
@@ -35,12 +35,16 @@ import qualified Spago.PackageSet              as PackageSet
 --       * if yes, download the tar archive and copy it to global and then local cache
 --       * if not, run a series of git commands to get the code, and copy to local cache
 fetchPackages
-  :: (HasLogFunc env, HasJobs env, HasGlobalCache env, HasPackageSet env)
+  :: (HasLogFunc env, HasJobs env, HasGlobalCache env, HasPackageSet env, HasGlobalOffline env)
   => [(PackageName, Package)]
   -> RIO env ()
 fetchPackages allDeps = do
   logDebug "Running `fetchPackages`"
 
+  internetAccess <- view (the @GlobalOffline)
```

Comment:
This would prevent anyone from getting in this code path at all - however there are some cases where we don't need to reach for the internet at all, e.g. when all packages are in cache. About 10 lines from here we bind `depsToFetch` which contains the list of things that we actually have to download (i.e. we filter out all the things in cache). We can fail then if `nOfDeps > 0` and the flag is set